dapp/src/utils/web3.js

```diff
@@ -44,7 +52,36 @@ export function truncateAddress(address) {
   return `${address.slice(0, 6)}...${address.slice(-4)}`
 }
 
+export function trackOUSDInMetamask(ousdAddress) {
+  web3.currentProvider.sendAsync(
+    {
+      method: 'metamask_watchAsset',
+      params: {
+        type: 'ERC20',
+        options: {
+          address: ousdAddress,
+          symbol: 'OUSD',
+          decimals: 18,
+          image: 'https://ousd.com/images/ousd-token-icon.svg',
+        },
+      },
+    },
+    console.log
+  )
+}
+
+/* status of token wallets and OUSD:
+ * https://docs.google.com/spreadsheets/d/1bunkxBxfkAVz9C14vAFH8CZ53rImDNHTXp94AOEjpq0/edit#gid=1608902436
+ */
+export function providersNotAutoDetectingOUSD() {
+  return ['metamask', 'trust', 'alphawallet', 'mist', 'parity']
```

Comment:
@micahalcorn added this bit of functionality where we whitelist the providers where we show the "add OUSD modal". We can cross them out from here when the wallets pick up OUSD, or add new ones if we see in analytics that some provider is popular and isn't auto detecting OUSD.